In [66]:
import pandas as pd
import matplotlib
import plotly
import matplotlib.pyplot as plt
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [3]:
# Cargamos las actividades de las ciudades
df_actividades_berlin = pd.read_csv("../../data/raw/actividades_berlin.csv")
df_actividades_viena = pd.read_csv("../../data/raw/actividades_viena.csv")

# Cargamos los alojamiento de las ciudades
df_airbnb_berlin = pd.read_csv("../../data/raw/AirBnB_Berlin.csv")
df_airbnb_viena = pd.read_csv("../../data/raw/AirBnB_Viena.csv")

# Cargamos los vuelos de Berlin
df_berlin_ida = pd.read_csv("../../data/eda/berlin_ida_directo.csv")
df_berlin_vuelta = pd.read_csv("../../data/eda/berlin_vuelta_directo.csv")

# Cargamos los vuelos de Viena
df_viena_ida = pd.read_csv("../../data/eda/viena_ida_directo.csv")
df_viena_vuelta = pd.read_csv("../../data/eda/viena_vuelta_directo.csv")


In [4]:
# Filtramos los vuelos más económicos para ida y vuelta de Berlín y Viena

# Berlín
berlin_ida_min = df_berlin_ida['Precio Total (EUR)'].min()
berlin_vuelta_min = df_berlin_vuelta['Precio Total (EUR)'].min()
berlin_vuelos_total_min = berlin_ida_min + berlin_vuelta_min

# Viena
viena_ida_min = df_viena_ida['Precio Total (EUR)'].min()
viena_vuelta_min = df_viena_vuelta['Precio Total (EUR)'].min()
viena_vuelos_total_min = viena_ida_min + viena_vuelta_min

print(f"Los vuelos más baratos para Berlín contando ida y vuelta cuesta esto: {berlin_vuelos_total_min} y pra Viena cuesta esto: {viena_vuelos_total_min}.(Contando las 5 personas)")


Los vuelos más baratos para Berlín contando ida y vuelta cuesta esto: 3627.2 y pra Viena cuesta esto: 2679.7.(Contando las 5 personas)


In [5]:
# Calculamos alojamiento mínimo para el grupo (5 personas)
berlin_alojamiento_min = df_airbnb_berlin['Precio Total'].min()
viena_alojamiento_min = df_airbnb_viena['Precio Total'].min()

print(f"Los alojamientos más baratos para Berlín contando ida y vuelta cuesta esto: {berlin_alojamiento_min} y pra Viena cuesta esto: {viena_alojamiento_min}.(Contando las 5 personas)")


Los alojamientos más baratos para Berlín contando ida y vuelta cuesta esto: 215 EUR y pra Viena cuesta esto: 103 EUR.(Contando las 5 personas)


In [6]:
# Encontramos la actividad con el costo mínimo en Berlín y Viena
actividad_berlin_min = df_actividades_berlin.loc[df_actividades_berlin['precio'].idxmin()]
actividad_viena_min = df_actividades_viena.loc[df_actividades_viena['precio'].idxmin()]

print(f"La actividad más barata para Berlín es: {actividad_berlin_min[['titulo', 'precio']]} La activiada más barata para Viena es: {actividad_viena_min[['titulo', 'precio']]}")


La actividad más barata para Berlín es: titulo    Free tour por Berlín
precio                     0.0
Name: 1, dtype: object La activiada más barata para Viena es: titulo    Free tour por Viena
precio                    0.0
Name: 1, dtype: object


In [80]:
#Calcular el total
berlin_total = berlin_vuelos_total_min + berlin_alojamiento_min
viena_total = viena_vuelos_total_min + viena_alojamiento_min
# Lo que cuesta cada cosa
print(f"El precio total de ir a Berlín es: {berlin_total}, y el de Viena es: {viena_total}. La diferencia de precio entre ellos es de {berlin_total-viena_total}")


El precio total de ir a Berlín es: 3842.2, y el de Viena es: 2782.7. La diferencia de precio entre ellos es de 1059.5


In [70]:
productos = ['Viena', 'Berlín']
categorias = ['Vuelos', 'Alojamientos']
precios_vuelos = [2679.7, 3627.2]
precios_alojamientos = [103, 215]


fig = make_subplots(rows=1, cols=2, subplot_titles=("Precios de Viena", "Precios de Berlín"))

fig.add_trace(go.Bar(x=categorias, y=[precios_vuelos[0], precios_alojamientos[0]], name="Viena"), row=1, col=1)

# Datos para Berlín
fig.add_trace(go.Bar(x=categorias, y=[precios_vuelos[1], precios_alojamientos[1]], name="Berlín", marker_color="orange"), row=1, col=2)

fig.update_layout(
    title_text="Comparación Interactiva de Precios por Categoría en Viena y Berlín",
    xaxis_title="Categorías",
    yaxis_title="Precio (€)",
    showlegend=False
)

fig.show()



### Conclusión  
  
Si nos guiamos unicamente por los precios Viena es notablemente más barato, concretamente 1059.50 € , Recuerdo que los precios son para las 5 personas